python/setup.py

```diff
@@ -227,7 +227,7 @@ def get_packages(self):
 if setup_spec.type == SetupType.RAY:
     pandas_dep = "pandas >= 1.3"
     numpy_dep = "numpy >= 1.20"
-    pyarrow_dep = "pyarrow >= 6.0.1"
+    pyarrow_dep = "pyarrow >= 6.0.1,<18"
```

Comment:
you did not add the darwin constraint here..

if some user wants to use the latest pyarrow with the next release, they will come back to you and complain..

Reply:
good catch, thanks. had this change committed but forgot to push it...
https://github.com/ray-project/ray/pull/48446